gremlin-go/driver/result.go

```diff
@@ -21,18 +21,149 @@ package gremlingo
 
 // AN-968 Finish Result implementation.
 
-import "fmt"
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+)
 
 // Result Struct to abstract the Result and provide functions to use it.
 type Result struct {
 	result interface{}
 }
 
-// AsString returns the current Result formatted as a string.
+func newResult(result interface{}) *Result {
+	return &Result{result}
+}
+
```

Comment:
I think you can remove this, there isn't much of an advantage.. Ex:
`foo := newResult(data)`
vs.
`foo := &Result(data)`

```suggestion
```